colorama/ansitowin32.py

```diff
@@ -41,6 +42,8 @@ def __getattr__(self, name):
     def write(self, text):
         self.__convertor.write(text)
 
+    def isatty(self):
+        return is_a_tty(self.__wrapped)
```

Comment:
This is interesting. Maybe we don't even need the functions `is_a_tty` and `is_stream_closed` above, and they should instead be methods of `StreamWrapper`, so that calling these methods will return what colorama "thinks" instead of the original. i.e. put the code here instead, and call the methods instead of the functions later in the code. What do you think?